conformance/vfs_assignement_test.go

```diff
@@ -0,0 +1,189 @@
+
+			Eventually(func() sriovv1.Interfaces {
+				nodeState, err := clients.SriovNetworkNodeStates(operatorNamespace).Get(testNode, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				return nodeState.Spec.Interfaces
+			}, 1*time.Minute, 1*time.Second).Should(ContainElement(MatchFields(
+				IgnoreExtras,
+				Fields{
+					"Name":     Equal(intf.Name),
+					"NumVfs":   Equal(2),
+					"VfGroups": ContainElement(sriovv1.VfGroup{ResourceName: "testresource", DeviceType: "netdevice", VfRange: "0-1"}),
+				})), "Error SriovNetworkNodeState doesn't conatin reqired elements")
+			Eventually(func() int64 {
+				testedNode, err := clients.Nodes().Get(testNode, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				resNum, _ := testedNode.Status.Capacity["openshift.io/testresource"]
+				capacity, _ := resNum.AsInt64()
+				return capacity
+			}, 3*time.Minute, time.Second).Should(Equal(int64(2)), "Error discovered less than 2 VFs")
+
+			networkConfig = &sriovv1.SriovNetwork{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: "testnetwork",
+					Namespace:    operatorNamespace,
+				},
+				Spec: sriovv1.SriovNetworkSpec{
+					NetworkNamespace: namespaces.Test,
+					Vlan:             0,
+					IPAM: `{ "type": "host-local","subnet": "10.10.10.0/24","rangeStart": 
+					"10.10.10.171","rangeEnd": "10.10.10.181","routes": [{ "dst": "0.0.0.0/0" }],"gateway": "10.10.10.1"}`,
+					ResourceName: nodePolicyConfig.Spec.ResourceName,
+				},
+			}
+			err = clients.Create(context.TODO(), networkConfig)
+			Expect(err).ToNot(HaveOccurred(), "Error to create SriovNetwork")
+			netAttDef := &netattdefv1.NetworkAttachmentDefinition{}
+			Eventually(func() error {
+				return clients.Get(context.TODO(), runtimeclient.ObjectKey{Name: networkConfig.Name, Namespace: namespaces.Test}, netAttDef)
+			}, 30*time.Second, 1*time.Second).ShouldNot(HaveOccurred(), "Error to detect NetworkAttachmentDefinition")
+		})
+
+		Context("Virtual Functions", func() {
+			// 21396
+			It("The VFs should be released once the pod deleted and can be used by the new created pods", func() {
+				By("Create first Pod which consumes all available VFs")
+				testPodA := &v1core.Pod{
+					ObjectMeta: metav1.ObjectMeta{
+						GenerateName: "testvfsa",
+						Namespace:    namespaces.Test,
+						Annotations:  map[string]string{"k8s.v1.cni.cncf.io/networks": fmt.Sprintf("%s,%s", networkConfig.Name, networkConfig.Name)},
+					},
+					Spec: v1core.PodSpec{
+						Containers: []v1core.Container{
+							{
+								Name:    "testvfcontainer",
+								Image:   "busybox",
+								Command: []string{"sleep", "10h"},
+							},
+						},
+					},
+				}
+
+				runningPodA, err := clients.Pods(testPodA.Namespace).Create(testPodA)
+				Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("Error to create pod %s", testPodA.Name))
+				By("Checking that first Pod is in Running state")
+				whaitForPodStatus(runningPodA, v1core.PodRunning, 3*time.Minute)
```

Comment:
waitForPodStatus. Also, some coordination with https://github.com/openshift/sriov-tests/pull/18 will be required as both have a pod waiting logic. My vote goes to a local Eventually function, like we do here https://github.com/openshift/sriov-tests/pull/18/files#diff-c17e18d66aee5ed11707fbfcb5601f92R284